app/assets/javascripts/shared/appState.coffee

```diff
@@ -0,0 +1,22 @@
+angular.module("SharedServices").service("AppState", ["$cookieStore", ($cookieStore) ->
+    user = {}
+    loggedIn = false
+
+    # User state
+    Object.defineProperty(this, "user", {
+        get: () -> user
+    })
+
+    this.setUserId = (id) -> user.id = id
+    this.setUserAvatar = (avatar) -> user.avatar = avatar
+    this.setUserName = (name) -> user.username = name
+
+    # Login state
+    Object.defineProperty(this, "loggedIn", {
+        get: () -> loggedIn
+    })
+
+    this.login = () -> loggedIn = false
```

Comment:
yea, it's unused so I'm not gonna change it right now. It's removed in the next patch.